Make sure highlighted option is distinguishable in forced colors mode (#512)
alphagov/accessible-autocomplete: README.md

`inputValue` is a function that receives one argument, the currently selected suggestion. It returns the string value to be inserted into the input.

> `suggestion` is a function that receives one argument, a suggestion to be displayed. It is used when rendering suggestions, and should return a string, which can contain HTML.

Reviewer: Sorry in advance if I've misunderstood - I'm guessing the suggestion is displayed when the user types into a text-field? (Let me know if this is one of those cases where using the passive voice is preferable to writing something contorted.)

Maybe something like this could work:

> `suggestion` is a function that receives one argument, a suggestion for the dropdown to display. The autocomplete uses this function when rendering suggestions. `suggestion` should return a string, which can contain HTML.

Author: I haven't added this line in this PR – just split it up. However, whilst we're here, I agree it could be improved and your suggestion's a really good start.

I think it'd probably make sense to swap the order around – at the minute we talk about what the function takes as input, what it's for, and what it should output… which seems muddled.

What about something like…

> `suggestion` can be used to customise how suggestions are rendered in the dropdown. It must be a function which will be called with one argument, which will be the suggestion. The function must return a string, which can contain HTML.

Author: The more I pick at this the more I want to rewrite the whole section on `templates` – possibly trying to cover off #420 at the same time.

I feel like that's starting to really stretch the scope of this PR, so how would you feel about me adding the new line in without making changes to the existing content, and creating a new issue or PR to tackle the surrounding docs?

Author: @EoinShaughnessy any objections to leaving further improvements to the existing docs out for now?

Reviewer: @36degrees No objections! Sorry for late reply. Give me a shout whenever we want to tackle the follow-up issue.

> :warning: **Caution:** because this function allows you to output arbitrary HTML, you should [make sure it's trusted](https://en.wikipedia.org/wiki/Cross-site_scripting), and accessible.

Reviewer: What does 'arbitrary' mean here? Random?

I get that it's the HTML that needs to be trusted and accessible, but that might not be totally clear at first glance.

Author: This line also isn't new, just split up.

It's a bit wordier, but what about…

>  To prevent [cross-site scripting vulnerabilities](https://en.wikipedia.org/wiki/Cross-site_scripting) ensure that the string returned by this function contains only trusted or sanitised input. You should also ensure that any additional HTML markup is accessible.

Reviewer: Yeah, that sounds good. Maybe just the lightest of tweaks:

> To prevent [cross-site scripting vulnerabilities](https://en.wikipedia.org/wiki/Cross-site_scripting), make sure the string returned by this function contains only trusted or sanitised input. You should also make sure any additional HTML markup is accessible.

(Sorry if this seems finicky - we're encouraged to write 'make sure' (plainer alternative) instead of 'ensure', and remove redundant _that_s.)

> If your template includes child elements with defined foreground or background colours, you should check that they display correctly in forced colors modes such as Windows high contrast mode.

Reviewer: ```suggestion
If your template includes child elements with defined foreground or background colours, you should check they display correctly in forced colors modes. For example, Windows high contrast mode.
```
